### PR TITLE
azurerm_data_factory_trigger_schedule supports schedule and description

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccDataFactoryTriggerSchedule_basic
=== PAUSE TestAccDataFactoryTriggerSchedule_basic
=== CONT  TestAccDataFactoryTriggerSchedule_basic
--- PASS: TestAccDataFactoryTriggerSchedule_basic (213.63s)
=== RUN   TestAccDataFactoryTriggerSchedule_complete
=== PAUSE TestAccDataFactoryTriggerSchedule_complete
=== CONT  TestAccDataFactoryTriggerSchedule_complete
--- PASS: TestAccDataFactoryTriggerSchedule_complete (388.00s)
=== RUN   TestAccDataFactoryTriggerSchedule_schedule
=== PAUSE TestAccDataFactoryTriggerSchedule_schedule
=== CONT  TestAccDataFactoryTriggerSchedule_schedule
--- PASS: TestAccDataFactoryTriggerSchedule_schedule (213.27s)
PASS
```

### CODE REVIEW

File: website/docs/r/data_factory_trigger_schedule.html.markdown [hunk 1]
```diff
@@ -53,18 +53,44 @@ The following arguments are supported:
 
 * `pipeline_name` - (Required) The Data Factory Pipeline name that the trigger will act on.
 
+* `description` - (Optional) The Schedule Trigger's description.
+
+* `schedule` - (Optional) The recurrence schedule for the trigger. A trigger with a specified frequency value alters its recurrence based on a recurrence schedule. The schedule property contains modifications for the recurrence that are based on minutes, hours, weekdays, month days, and week number. A `schedule` block as documented below.
```
Comment (stephybun) [inline website/docs/r/data_factory_trigger_schedule.html.markdown:58]:
```suggestion
* `schedule` - (Optional) A `schedule` block as defined below, which further specifies the recurrence schedule for the trigger. A schedule is capable of limiting or increasing the number of trigger executions specified by the `frequency` and `interval` properties.
```

File: website/docs/r/data_factory_trigger_schedule.html.markdown [hunk 2]
```diff
@@ -53,18 +53,44 @@ The following arguments are supported:
...
 
 * `pipeline_parameters` - (Optional) The pipeline parameters that the trigger will act upon.
 
 * `annotations` - (Optional) List of tags that can be used for describing the Data Factory Schedule Trigger.
 
+---
+
+`schedule` supports the following:
+
+* `days_of_month` - (Optional) 	Day of the month on which the trigger runs. The value can be specified with a monthly frequency only.
```
Comment (stephybun) [inline website/docs/r/data_factory_trigger_schedule.html.markdown:76]:
```suggestion
* `days_of_month` - (Optional) Day(s) of the month on which the trigger is scheduled. This value can be specified with a monthly frequency only.
```

File: website/docs/r/data_factory_trigger_schedule.html.markdown [hunk 3]
```diff
@@ -53,18 +53,44 @@ The following arguments are supported:
...
+
+* `days_of_week` - (Optional) Days of the week the trigger runs. The value can be specified only with a weekly frequency.
+
+* `hours` - (Optional) Hours of the day at which the trigger runs.
```
Comment (stephybun) [inline website/docs/r/data_factory_trigger_schedule.html.markdown:80]:
```suggestion
* `hours` - (Optional) Hours of the day on which the trigger is scheduled.
```

File: website/docs/r/data_factory_trigger_schedule.html.markdown [hunk 4]
```diff
@@ -53,18 +53,44 @@ The following arguments are supported:
...
+
+* `minutes` - (Optional) Minutes of the hour at which the trigger runs.
+
+* `monthly_occurrence` - (Optional) Days of the month on which the trigger runs. The value can be specified with a monthly frequency only. A `monthly_occurrence` block as documented below.
+
+---
+
+`monthly_occurrence` supports the following:
```
Comment (stephybun) [inline website/docs/r/data_factory_trigger_schedule.html.markdown:88]:
```suggestion
A `monthly_occurrence` block supports the following:
```

File: website/docs/r/data_factory_trigger_schedule.html.markdown [hunk 5]
```diff
@@ -53,18 +53,44 @@ The following arguments are supported:
...
+
+* `day` - (Required) The day of the week on which the trigger runs. For example, a `monthly_occurrence` property with a day value of `Sunday` means every Sunday of the month.
+
+* `occurrence` - (Optional) The occurrence of the specified day during the month. For example, a `monthly_occurrence` property with day and occurrence values of `Sunday, -1` means the last Sunday of the month.
```
Comment (stephybun) [inline website/docs/r/data_factory_trigger_schedule.html.markdown:92]:
```suggestion
* `occurrence` - (Optional) The occurrence of the specified day during the month. For example, a `monthly_occurrence` property with `day` and `occurrence` values of `Sunday, -1` means the last Sunday of the month.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @ms-henglu, tests look good, just a few minor remarks and some suggestions for the docs. Once those are addressed then this LGTM 👍🏼

File: internal/services/datafactory/data_factory_trigger_schedule_resource.go [hunk 6]
```diff
@@ -53,6 +53,86 @@ func resourceDataFactoryTriggerSchedule() *pluginsdk.Resource {
 				ValidateFunc: validate.DataFactoryName(),
 			},
 
+			"description": {
```
[the remark this answers is not in the document]
Comment (stephybun) [reply internal/services/datafactory/data_factory_trigger_schedule_resource.go:56]: Disregard this comment, I think github/browser is messing with the appearance of the formatting.

File: internal/services/datafactory/data_factory_trigger_schedule_resource.go [hunk 7]
```diff
@@ -53,6 +53,86 @@ func resourceDataFactoryTriggerSchedule() *pluginsdk.Resource {
...
+						"minutes": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							Elem: &pluginsdk.Schema{
+								Type:         pluginsdk.TypeInt,
+								ValidateFunc: validation.IntBetween(0, 60),
+							},
+						},
+
+						"monthly_occurrence": {
```
Comment (katbyte) [inline internal/services/datafactory/data_factory_trigger_schedule_resource.go:109]: i think we can remove occurrence here for a bit better UX
```suggestion
						"monthly": {
```

File: internal/services/datafactory/data_factory_trigger_schedule_resource.go [hunk 8]
```diff
@@ -53,6 +53,86 @@ func resourceDataFactoryTriggerSchedule() *pluginsdk.Resource {
...
+							MinItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"day": {
+										Type:         pluginsdk.TypeString,
+										Required:     true,
+										ValidateFunc: validation.IsDayOfTheWeek(false),
+									},
+
+									"occurrence": {
```
Comment (katbyte) [inline internal/services/datafactory/data_factory_trigger_schedule_resource.go:121]: and occurrence is not clear, how about
```suggestion
									"week": {
```

File: internal/services/datafactory/data_factory_trigger_schedule_resource.go [hunk 9]
```diff
@@ -53,6 +53,86 @@ func resourceDataFactoryTriggerSchedule() *pluginsdk.Resource {
...
+							},
+						},
+
+						"monthly_occurrence": {
+							Type:     pluginsdk.TypeList,
+							Optional: true,
+							MinItems: 1,
+							Elem: &pluginsdk.Resource{
+								Schema: map[string]*pluginsdk.Schema{
+									"day": {
```
Comment (katbyte) [inline internal/services/datafactory/data_factory_trigger_schedule_resource.go:115]: would this be more clear as 
```suggestion
									"weekday": {
```

**Comment by ms-henglu** [comment]
@katbyte @stephybun Hi all, thanks for code review!
I've added a commit according to your suggestions, please take another look, thanks!

**Review by katbyte — APPROVED** [review]
Thanks @ms-henglu - LGTM 🍰

**Pull request merged**